### DISCUSSION TITLE
Xcode errors stating the external macro for the reducer cannot be found

### DISCUSSION THREAD

**Discussion by holzerryan** [pointfreeco/swift-composable-architecture#3401] [General]
I'm having an issue with Xcode where I'm seeing the following error in many places.
External macro implementation type 'ComposableArchitectureMacros.ReducerMacro' could not be found for macro 'Reducer()'

I've added the swift package for the composable architecture and am using version 1.14.0.  My project compiles and launches in a simulator and there are no issues...but xcode continues to display these errors in the UI.  It makes it very difficult to write code and debug problems when there are actual issues.  

I'm currently using Xcode version 15.4, but I've tested this with other versions of Xcode and am experiencing the same problem.  I've tried switching to Xcode version 15.2, and 15.3.  I have also completely uninstalled Xcode from my local machine and re-installed, but the issue continues. (assuming this is a problem with my local machine)

I'm attaching a screen shot of the issue as a picture is worth a 1000 words.
![Screenshot 2024-09-20 at 11 06 10 AM](https://github.com/user-attachments/assets/f4a9409e-8d65-4989-b4ff-45cb76cc9455)

Is anyone else having this problem?

**Comment by mbrandonw** [comment]
Hi @holzerryan, it has been reported that this happens when using custom build configurations besides the standard Debug and Release. Is that your situation? If so, then switching back to the standard configuration names and using schemes instead for build customization would probably get you back into good shape. I'd also recommend filing a Feedback with Apple so that they are aware of this bug in Xcode.

**Reply by holzerryan** [reply]
Yes, I am using custom build configurations.  Thank you for the suggestion.  I will revert back to the standard Debug and Release configurations and see if that fixes my problem.  I appreciate the quick response and advice.

**Comment by holzerryan** [comment]
I'm unable to remove the custom configurations because we need to compile the app with multiple bundle ids.  Is there a way to package the app with an alternate bundle id using only a scheme and not with a custom configuration?

**Reply by mbrandonw** [reply]
The only thing that comes to mind is to keep the standard Debug and Release configurations and then add your custom configurations on top of that. Outside of that I'm not really sure, it may be more helpful to ask this question in a more general Apple platform setting, such as on Apple's developer forums. And of course this is a really annoying Xcode bug, and so you should file a Feedback with Apple about it.

**Reply by holzerryan** [reply]
Actually, I was able to work around the issue by using the custom configurations only for CI build pipelines.  All local development work on the project uses schemes that rely on the Debug and Release configuration.  I updated logic in the app to use environment variables set in the schemes instead of using values passed by the build settings and custom configurations.  

Thank you for the feedback and recommendation.  It moved me past my problem and now I'm back to writing more code with TCA.
